commit ded4f96bdbcc448d5586f4d95f99a36484c7202f
Author: John Johansen <john.johansen@canonical.com>
Date:   Mon May 23 11:52:18 2016 -0700

    UBUNTU: SAUCE: apparmor: Fix __label_update proxy comparison test
    
    The comparing the proxy pointer, not the address of the labels proxy pointer.
    
    BugLink: http://bugs.launchpad.net/bugs/1615878
    
    Signed-off-by: John Johansen <john.johansen@canonical.com>
    Signed-off-by: Leann Ogasawara <leann.ogasawara@canonical.com>

diff --git a/security/apparmor/label.c b/security/apparmor/label.c
index 21c9d6f..c453fc8 100644
--- a/security/apparmor/label.c
+++ b/security/apparmor/label.c
@@ -1993,7 +1993,7 @@ static struct aa_label *__label_update(struct aa_label *label)
 	write_lock_irqsave(&ls->lock, flags);
 	label_for_each(i, label, p) {
 		new->vec[i.i] = aa_get_newest_profile(p);
-		if (&new->vec[i.i]->label.proxy != &p->label.proxy)
+		if (new->vec[i.i]->label.proxy != p->label.proxy)
 			invcount++;
 	}
 
